MOD: Refactor task_navigator.py

okahilak opened the pull request:
- Some steps towards decoupling GUI and domain logic, namely, by adding
  classes Navigation and Tracker, which take some of the responsibility
  previously handled by NeuronavigationPanel.

- This is done to prepare to do some changes in the GUI, such as
  moving "Navigate" button elsewhere -- previously, it was hard
  to do due to the elements in the GUI being tightly coupled with the
  implementation of navigation.

- Also some other minor clean-up and style improvements, such as
  using more consistent and more descriptive variable names

tfmoraes commented:
@okahilak if I activate **Transcranial Magnetic Stimulation Mode" and after that I try to close InVesalius it gives me this error:

```shell
Traceback (most recent call last):
  File "/home/thiago/Sources/github/invesalius3/invesalius/gui/frame.py", line 408, in OnClose
    Publisher.sendMessage('Close Project')
  File "/home/thiago/Sources/github/invesalius3/invesalius/pubsub/pub.py", line 74, in sendMessage
    Publisher.sendMessage(topicName, **msgdata)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/publisher.py", line 216, in sendMessage
    topicObj.publish(**msgData)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/topicobj.py", line 452, in publish
    self.__sendMessage(msgData, topicObj, msgDataSubset)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/topicobj.py", line 482, in __sendMessage
    listener(data, self, allData)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/listener.py", line 237, in __call__
    cb(**kwargs)
  File "/home/thiago/Sources/github/invesalius3/invesalius/control.py", line 164, in OnShowDialogCloseProject
    self.ShowDialogCloseProject()
  File "/home/thiago/Sources/github/invesalius3/invesalius/control.py", line 300, in ShowDialogCloseProject
    self.CloseProject()
  File "/home/thiago/Sources/github/invesalius3/invesalius/control.py", line 392, in CloseProject
    Publisher.sendMessage('Close project data')
  File "/home/thiago/Sources/github/invesalius3/invesalius/pubsub/pub.py", line 74, in sendMessage
    Publisher.sendMessage(topicName, **msgdata)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/publisher.py", line 216, in sendMessage
    topicObj.publish(**msgData)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/topicobj.py", line 452, in publish
    self.__sendMessage(msgData, topicObj, msgDataSubset)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/topicobj.py", line 482, in __sendMessage
    listener(data, self, allData)
  File "/nix/store/8w5g0xkhvyh47w7mpdsfvhykq74i8m70-python3-3.8.10-env/lib/python3.8/site-packages/pubsub/core/listener.py", line 237, in __call__
    cb(**kwargs)
  File "/home/thiago/Sources/github/invesalius3/invesalius/gui/task_navigator.py", line 1121, in OnCloseProject
    self.ResetTrackerFiducials()
AttributeError: 'NeuronavigationPanel' object has no attribute 'ResetTrackerFiducials'
```

okahilak commented:
> @okahilak if I activate **Transcranial Magnetic Stimulation Mode" and after that I try to close InVesalius it gives me this error:

Thanks for the observation -- it should be fixed by the commit that I added. I tested different combinations of opening and closing the project, including the following:

- Opening an existing project and closing InVesalius
- Opening an existing project twice
- Opening an existing project, setting up navigation, and opening another project
- Opening an existing project, setting up navigation, starting the navigation, and opening another project while navigating

All the combinations seemed to work with the latest commit, except the last one. However, it looks like it doesn't work in the current main branch, either. I created an issue for that (#304).

@vhosouza The commit that I added should also address your concerns regarding doing all the necessary resetting steps in OnCloseProject. I wasn't able to delete Navigation and Tracker objects due to the other Panels triggering events in their own OnCloseProject functions that need the Navigation and Tracker objects to exist. However, I ended up re-initializing them, which should get them as close to a resetted state as possible.
